userapi/src/main/java/fk/prof/userapi/api/AggregatedProfileLoader.java

```diff
@@ -154,40 +175,41 @@ protected void loadSummaryFromInputStream(Future<AggregationWindowSummary> futur
         try {
             CheckedInputStream cin = new CheckedInputStream(in, checksum);
 
-            int magicNum = Deserializer.readVariantInt32(cin);
+            int magicNum = ProtoUtil.readVariantInt32(cin);
 
             if (magicNum !=  AggregationWindowSummarySerializer.SUMMARY_FILE_MAGIC_NUM) {
                 future.fail("Unknown file. Unexpected first 4 bytes");
                 return;
             }
 
             // read header
-            AggregatedProfileModel.Header parsedHeader = Deserializer.readCheckedDelimited(AggregatedProfileModel.Header.parser(), cin, "header");
+            AggregatedProfileModel.Header parsedHeader = ProtoUtil.buildProtoFromCheckedInputStream(AggregatedProfileModel.Header.parser(), cin, "header");
 
             // read traceCtx list
-            AggregatedProfileModel.TraceCtxNames traceNames = Deserializer.readCheckedDelimited(AggregatedProfileModel.TraceCtxNames.parser(), cin, "traceNames");
+            AggregatedProfileModel.TraceCtxNames traceNames = ProtoUtil.buildProtoFromCheckedInputStream(AggregatedProfileModel.TraceCtxNames.parser(), cin, "traceNames");
 
             // read profiles summary
             checksumReset(checksum);
             List<AggregatedProfileModel.ProfileWorkInfo> profiles = new ArrayList<>();
             int size = 0;
-            while((size = Deserializer.readVariantInt32(cin)) != 0) {
+            while((size = ProtoUtil.readVariantInt32(cin)) != 0) {
                 profiles.add(AggregatedProfileModel.ProfileWorkInfo.parseFrom(ByteStreams.limit(cin, size)));
             }
-            checksumVerify((int)checksum.getValue(), Deserializer.readVariantInt32(in), "checksum error profileWorkInfo");
+            checksumVerify((int)checksum.getValue(), ProtoUtil.readVariantInt32(in), "checksum error profileWorkInfo");
 
             // read work specific samples
             Map<AggregatedProfileModel.WorkType, AggregationWindowSummary.WorkSpecificSummary> summaryPerTrace = new HashMap<>();
 
             // cpu_sampling
-            AggregatedProfileModel.TraceCtxDetailList traceDetails = Deserializer.readCheckedDelimited(AggregatedProfileModel.TraceCtxDetailList.parser(), cin, "cpu_sample traceDetails");
+            AggregatedProfileModel.TraceCtxDetailList traceDetails = ProtoUtil.buildProtoFromCheckedInputStream(AggregatedProfileModel.TraceCtxDetailList.parser(), cin, "cpu_sample traceDetails");
             summaryPerTrace.put(AggregatedProfileModel.WorkType.cpu_sample_work, new AggregationWindowSummary.CpuSampleSummary(traceDetails));
 
             AggregationWindowSummary summary = new AggregationWindowSummary(parsedHeader, traceNames, profiles, summaryPerTrace);
 
             future.complete(summary);
         }
         catch (IOException e) {
+            profileSummaryLoadFailureCounter.mark();
```

Comment:
Why catch the exception here? Let this method throw exception and can be handled by the caller, which anyways does that today (failure metric and future are both handled in the caller)

Reply:
yup. now all provate methods just do their work and throw any exception occured. Caller now handles everything else.